include/engine/douglas_peucker.hpp

```diff
@@ -44,14 +44,16 @@ const constexpr auto DOUGLAS_PEUCKER_THRESHOLDS_SIZE =
 // Input is vector of pairs. Each pair consists of the point information and a
 // bit indicating if the points is present in the generalization.
 // Note: points may also be pre-selected*/
-void douglasPeucker(std::vector<SegmentInformation>::iterator begin,
-                    std::vector<SegmentInformation>::iterator end,
-                    const unsigned zoom_level);
+std::vector<util::FixedPointCoordinate>
+douglasPeucker(std::vector<util::FixedPointCoordinate>::const_iterator begin,
+               std::vector<util::FixedPointCoordinate>::const_iterator end,
+               const unsigned zoom_level);
```

Comment:
To circumvent creating a vector for each route step, the whole leg geometry is saved in one large vector and each step only saves indices into that. Technically decided now that per-step simplification is probably not a good idea, but this is a left-over from the previous design.